### ISSUE TITLE
iOS VoiceOver disrupts the lifecycle of platform views

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#73716]
Internal: b/178003647

Reproduction with Youtube API + Webview Flutter plugin:

```dart
import 'dart:async';
import 'dart:convert';

import 'package:flutter/material.dart';
import 'package:webview_flutter/webview_flutter.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatelessWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(title, style: TextStyle(fontFamily: 'ProductSans')),
      ),
      body: Center(
        child: TextButton(
          onPressed: () => _launchVideo(context), child: Text('Launch Video'),
        ),
      ),
    );
  }

  Future<void> _launchVideo(BuildContext context) async {
    await Navigator.push(
        context,
        MaterialPageRoute(
          fullscreenDialog: true,
          builder: (context) => _buildPopover(context),
        ));
  }

  Widget _buildPopover(BuildContext context) {
    return OrientationBuilder(
      builder: (context, orientation) {
        final isPortrait = orientation == Orientation.portrait;
        return Scaffold(
          appBar: isPortrait
              ? AppBar(
            title: Text('Video', ),
            leading: IconButton(
              icon: Icon(Icons.close),
              onPressed: () async {
                await Navigator.of(context).maybePop();
              },
            ),
          )
              : null,
          body: Container(
            decoration: BoxDecoration(
              color: Colors.black,
            ),
            child: SafeArea(child: YoutubePlayerWidget()),
          ),
        );
      },
    );
  }
}

const _userAgent =
    'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36';
const _aspectRatio = 16 / 9;

/// A youtube player widget which interacts with the underlying webview
/// to play YouTube videos.
class YoutubePlayerWidget extends StatefulWidget {
  @override
  _YoutubePlayerWidgetState createState() => _YoutubePlayerWidgetState();
}

class _YoutubePlayerWidgetState extends State<YoutubePlayerWidget> {
  @override
  Widget build(BuildContext context) {
    return AspectRatio(
        aspectRatio: _aspectRatio,
        child: IgnorePointer(
          ignoring: false,
          child: WebView(
            key: widget.key,
            userAgent: _userAgent,
            initialUrl: _player,
            javascriptMode: JavascriptMode.unrestricted,
            initialMediaPlaybackPolicy: AutoMediaPlaybackPolicy.always_allow,
          ),
        ));
  }

  String get _player {
    final _player = '''
    <!DOCTYPE html>
    <html>
    <head>
        <style>
            html,
            body {
                margin: 0;
                padding: 0;
                background-color: #000000;
                overflow: hidden;
                position: fixed;
                height: 100%;
                width: 100%;
            }
        </style>
        <meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no'>
    </head>
    <body>
        <div id="player"></div>
        <script>
            var tag = document.createElement('script');
            tag.src = "https://www.youtube.com/iframe_api";
            var firstScriptTag = document.getElementsByTagName('script')[0];
            firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
            var player;
            var timerId;
            function onYouTubeIframeAPIReady() {
                player = new YT.Player('player', {
                    height: '100%',
                    width: '100%',
                    videoId: '6wnPeqh4m2k',
                    host: 'https://www.youtube.com',
                    playerVars: {
                        'controls': 1,
                        'playsinline': 1,
                        'enablejsapi': 1,
                        'fs': 0,
                        'rel': 0,
                        'showinfo': 0,
                        'iv_load_policy': 3,
                        'modestbranding': 1,
                        'cc_load_policy': 0,
                        'cc_lang_pref': 'en',
                        'autoplay': true
                    },
                    events: {
                        onError: (error) => Errors.postMessage(error.data)
                    },
                });
            }
        </script>
    </body>
    </html>
    ''';
    return 'data:text/html;base64,'
        '${base64Encode(const Utf8Encoder().convert(_player))}';
  }
}
```

Steps:
1. Tap launch video.
2. After the video starts playing, enable picture-in-picture mode by tapping on this icon: 

![image](https://user-images.githubusercontent.com/6687929/104217146-dfaedc00-53ef-11eb-8074-15ff71e00db1.png)

3. Tap close button.

Observe that the video stops and the platform view is gone.

Now enable voiceover and repeat the same steps. You will see that after tapping close, the video continues to play. The `dispose` logic of platform view is not executed correctly with voiceover on.

**Comment by darshankawar** [comment]
<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 1.22.5, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-IN)
    • Flutter version 1.22.5 at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 7891006299 (5 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.9.3

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (3 available)
    • sdk gphone x86 64 arm64 (mobile)    • emulator-5554
      • android-x64 • Android 11 (API 30) (emulator)
    • Deepti 's iPhone (mobile)           •
      0554e90e563e81fd41bfd2ff9a2c6960595b9825 • ios         • iOS 12.4.9
    • iPhone SE (2nd generation) (mobile) • 6C85835D-FBFD-4AB3-8DE8-B4FAD35E5367
      • ios         • com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)
    ! Error: Deepti 's iPhone is busy: Copying cache files from device. Xcode
      will continue when Deepti 's iPhone is finished. (code -10)

! Doctor found issues in 1 category.
dhs@Dhss-MacBook-Pro ~ % flutter doctor -v
[✓] Flutter (Channel stable, 1.22.5, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-IN)
    • Flutter version 1.22.5 at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 7891006299 (5 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.9.3

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (3 available)
    • sdk gphone x86 64 arm64 (mobile)    • emulator-5554
      • android-x64 • Android 11 (API 30) (emulator)
    • Darshan's iphone (mobile)           •
      21150b119064aecc249dfcfe05e259197461ce23 • ios         • iOS 14.2
    • iPhone SE (2nd generation) (mobile) • 6C85835D-FBFD-4AB3-8DE8-B4FAD35E5367
      • ios         • com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)

! Doctor found issues in 1 category.


```

</details>

**Comment by goderbauer** [comment]
(Will require iOS 14 to reproduce on iPhone)

**Comment by mehmetf** [comment]
This is blocking an important feature launch for customer: mulligan.

**Comment by chunhtai** [comment]
unassign myself since i won't be able to get to now

**Comment by goderbauer** [comment]
@blasten Any change somebody familiar with the webview plugin could take a look at this?

**Comment by blasten** [comment]
Thanks @cyanglaz for taking a look!

**Comment by cyanglaz** [comment]
Updates:
I am able to reproduce the issue with the example app. I also set up an iOS app without flutter, which mimics how platform view and WebView works. And there is no issue with the iOS app. 
**So I can confirm this is an issue with flutter but not iOS.**
Will keep working on this.

**Comment by cyanglaz** [comment]
The fix has rolled into the framework at https://github.com/flutter/flutter/commit/b957144136c36f189f00876b71379a0a7cae7efc
